python/cudf/cudf/core/indexed_frame.py

```diff
@@ -410,7 +418,13 @@ def sort_index(
                 inds = idx.argsort(
                     ascending=ascending, na_position=na_position
                 )
-                out = self.take(inds)
+                out = self._gather(inds)
+                if isinstance(
+                    self, cudf.core.dataframe.DataFrame
+                ) and isinstance(
+                    self.columns, pd.core.indexes.multi.MultiIndex
+                ):
+                    out.columns = self.columns
```

Comment:
How was this being handled before? I don't see corresponding logic in take, were we missing it?

Reply:
The logic is in `DataFrame.take`. `.columns` attribute is specific to `DataFrame`, I believe.

Reply:
Oh oops I just looked in Frame and IndexedFrame and forgot to check DataFrame. OK yeah that makes sense. I guess all other uses of DataFrame._gather are internal-only, so we don't usually need to embed this behavior in that directly.

Reply:
For clarification, this was an offhand comment not intended to trigger any change here. The performance impact of setting columns is probably too high to try to do this right now.